cluster-autoscaler/metrics/liveness.go

```diff
@@ -76,3 +90,12 @@ func (hc *HealthCheck) UpdateLastActivity(timestamp time.Time) {
 		hc.lastActivity = timestamp
 	}
 }
+
+// UpdateLastSuccessfulRun updates last time of successful (i.e. not ending in error) activity
+func (hc *HealthCheck) UpdateLastSuccessfulRun(timestamp time.Time) {
+	hc.mutex.Lock()
+	defer hc.mutex.Unlock()
+	if timestamp.After(hc.lastSuccessfulRun) {
+		hc.lastSuccessfulRun = timestamp
```

Comment:
It probably doesn't matter all that much if we have reasonable timeouts, but maybe we should update lastActivity in here as well? My reasoning is that successfully finishing a loop means that it is running and we're about to do a 10s sleep. So theoretically if a loop took 9:55 we would fail in the middle of the sleep.